tuning/autotune.py

```diff
@@ -142,12 +143,75 @@ def get_candidate_mlir_path(self, candidate_id: int) -> Path:
         return self.candidates_dir / f"{candidate_id}.mlir"
 
     def get_candidate_spec_mlir_path(self, candidate_id: int) -> Path:
-        return self.candidates_dir / "configs" / f"{candidate_id}_spec.mlir"
+        return self.candidates_dir / "specs" / f"{candidate_id}_spec.mlir"
 
     def get_exe_format(self, path: Path) -> str:
         return f"./{path.as_posix()}"
 
 
+@dataclass
+class TuningClient(ABC):
+    @abstractmethod
+    def get_dispatch_compile_command(
+        self, candidate_tracker: CandidateTracker
+    ) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_dispatch_benchmark_command(self, candidate_tracker) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_model_compile_command(self, candidate_tracker) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_model_benchmark_command(self, candidate_tracker) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_compiled_dispatch_index(self, file_path: Path) -> int:
+        pass
```

Comment:
Why does the client decide the index?

Reply:
originally wanted to give users more freedom on file naming format